alpha/lib/model/DeliveryProfileLive.php

```diff
@@ -111,65 +111,55 @@ protected function initLiveStreamConfig()
+		$dcInMaintenance = $this->getIsMaintenanceFromCache($entryId);
+		KalturaLog::debug("Having requested-Server-Type of [$requestedServerType] and DC in maintenance of [$dcInMaintenance]");
+		$liveEntryServerNodes = array_filter($liveEntryServerNodes, function($esn) use ($dcInMaintenance, $requestedServerType){
+			if ($requestedServerType && $esn->getServerType() != intval($requestedServerType))
+				return false; // if request specific type then ignore all others 
+			$esn->sn = ServerNodePeer::retrieveActiveMediaServerNode(null, $esn->getServerNodeId());
+			if (!$esn->sn)
+				return false; // if the entry has no active server node ignore it
+			$esn->weight = ($esn->getServerType() == EntryServerNodeType::LIVE_PRIMARY) ? 10 : 0;
+			if ($esn->sn->getDc() == $dcInMaintenance)
+				$esn->weight -= 100; // if DC in maintenance then lower its priority
+			return true;
+		 });
+		if (empty($liveEntryServerNodes))
 			KExternalErrors::dieError(KExternalErrors::ENTRY_NOT_LIVE, "Entry [$entryId] is not broadcasting on stream type [$requestedServerType]");
-		}
 
-		if(!$this->liveStreamConfig->getUrl() && count($liveEntryServerNodes))
-		{
-			$liveEntryServerNode = array_shift($liveEntryServerNodes);
-			$serverNode = ServerNodePeer::retrieveActiveMediaServerNode(null, $liveEntryServerNode->getServerNodeId());
-			if($serverNode)
-			{
-				$this->liveStreamConfig->setUrl($this->getHttpUrl($serverNode));
-				$this->liveStreamConfig->setPrimaryStreamInfo($liveEntryServerNode->getStreams());
-			}
+		usort($liveEntryServerNodes, function ($a, $b) {return $a->weight - $b->weight;});
+
+		$liveEntryServerNode = array_shift($liveEntryServerNodes); // after sort first is the primary
+		$this->liveStreamConfig->setUrl($this->getHttpUrl($liveEntryServerNode->sn));
+		$this->liveStreamConfig->setPrimaryStreamInfo($liveEntryServerNode->getStreams());
+
+		$liveEntryServerNode = array_shift($liveEntryServerNodes);
+		if ($liveEntryServerNode) { // if list has another entry server node set it as backup
+			$this->liveStreamConfig->setBackupUrl($this->getHttpUrl($liveEntryServerNode->sn));
+			$this->liveStreamConfig->setBackupStreamInfo($liveEntryServerNode->getStreams());
 		}
-		
-		if(count($liveEntryServerNodes))
-		{
-			$liveEntryServerNode = array_shift($liveEntryServerNodes);
-			$serverNode = ServerNodePeer::retrieveActiveMediaServerNode(null, $liveEntryServerNode->getServerNodeId());
-			if ($serverNode)
-			{
-				$this->liveStreamConfig->setBackupUrl($this->getHttpUrl($serverNode));
-				$this->liveStreamConfig->setBackupStreamInfo($liveEntryServerNode->getStreams());
-			}
+	}
+
+	private function getIsMaintenanceFromCache($entryId)
+	{
+		$cache = kCacheManager::getSingleLayerCache(kCacheManager::CACHE_TYPE_PLAY_MANIFEST);
+		if ($cache) {
+			$result = json_decode($cache->get("MaintenanceDataCacheKey"), true);
+			KalturaLog::debug("Got maintenance data from cache: " . print_r($result, true));
```

Comment:
1.will add a check if this key even exists...
2.rename MaintenanceDataCacheKey to **Live-**MaintenanceData